WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModel.kt

```diff
@@ -903,6 +905,35 @@ class ReaderPostDetailViewModel @Inject constructor(
         }
     }
 
+    fun onUserNavigateFromComments() {
+        // reload post from DB and update UI state
+        val currentUiState: ReaderPostDetailsUiState? = (_uiState.value as? ReaderPostDetailsUiState)
+        currentUiState?.let {
+            findPost(currentUiState.postId, currentUiState.blogId)?.let { post ->
+                this.post = post
+                onUpdatePost(post)
+            }
+        }
+
+        // reload comments from DB and update comments snippet
+        // if it shows comments
+        if (_commentSnippetState.value !is CommentSnippetState.Failure &&
```

Comment:
I maybe would consider removing the check on Failure state. Test scenario I was thinking about:
- After turning on Airplane mode, go to a post detail where there are no comments yet
- You should get a No Network error in comments snippet
- Go to comments list screen and then turn off the ariplain mode
- Place a comment and come back to post detail
- Check that the snippet is not updated even if the conditions to make an update would be there

In my understanding removing that check on Failure would have alloud the previous scenario to be covered and the snippet updated (cannot think to possible side issues with that, but happy to get your thoughts of course in case I'm missing something 🙇 )

Reply:
Yep, you are right 👍 I thought for a bit, and also can't come up with a scenario where we want to keep the failure.